File: 백준/Silver/1590. 캠프가는 영식/캠프가는 영식.py
import sys
input=lambda:sys.stdin.readline().rstrip()
n,t=map(int,input().split())
result=[]
for _ in range(n):
    S,I,C=map(int,input().split())
    time=[S+(I*c) for c in range(C)]
    if time[-1]<t: continue
    start,end=0,C-1
    while start<=end:
        mid=(start+end)//2
        if time[mid]>=t:
            ans=mid
            end=mid-1
        else:
            start=mid+1
    result+=[time[ans]-t]
print(min(result) if result else -1)

File: 백준/Silver/1590. 캠프가는 영식/캠프가는 영식.cc
#include <algorithm>
#include <iostream>
#include <vector>
using namespace std;
int main(void) {
    int n, t, S, I, C, ans, start, end, mid;
    cin >> n >> t;
    vector<int> time, result;
    for (int i = 0; i < n; i++) {
        cin >> S >> I >> C;
        time = {};
        for (int k = 0; k < C; k++)
            time.push_back(S + I * k);
        if (time[C - 1] < t) continue;
        start = 0, end = C - 1;
        while (start <= end) {
            mid = (start + end) / 2;
            if (time[mid] >= t) {
                ans = mid;
                end = mid - 1;
            } else
                start = mid + 1;
        }
        result.push_back(time[ans] - t);
    }
    cout << (result.size() > 0 ? *min_element(result.begin(), result.end()) : -1);
    return 0;
}